diff -dur -x '*.pyc' -x '*.class' -x requires.txt Lib/site-packages/reportlab/lib/utils.py target/sphinx-tmp/jython/Lib/site-packages/reportlab/lib/utils.py
--- Lib/site-packages/reportlab/lib/utils.py	2014-04-17 21:28:27.975776000 +0200
+++ target/sphinx-tmp/jython/Lib/site-packages/reportlab/lib/utils.py	2014-04-17 21:29:29.424080771 +0200
@@ -516,7 +516,8 @@
 
 def _isPILImage(im):
     try:
-        return isinstance(im,Image.Image)
+        from PIL import Image
+        return isinstance(im, Image)
     except AttributeError:
         return 0
 
